homeassistant/components/spc/__init__.py

```diff
@@ -16,61 +17,80 @@
 CONF_API_URL = "api_url"
 
 DOMAIN = "spc"
-DATA_API = "spc_api"
 
 SIGNAL_UPDATE_ALARM = "spc_update_alarm_{}"
 SIGNAL_UPDATE_SENSOR = "spc_update_sensor_{}"
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_WS_URL): cv.string,
-                vol.Required(CONF_API_URL): cv.string,
-            }
+            vol.All(
+                cv.deprecated(CONF_WS_URL),
+                cv.deprecated(CONF_API_URL),
+            )
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
 async def async_setup(hass, config):
+    """Old way to set up SPC, import config."""
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_API_URL: conf[CONF_API_URL],
+                CONF_WS_URL: conf[CONF_WS_URL],
+            },
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
     """Set up the SPC component."""
 
-    async def async_upate_callback(spc_object):
+    config = entry.data
 
+    async def async_update_callback(spc_object):
         if isinstance(spc_object, Area):
             async_dispatcher_send(hass, SIGNAL_UPDATE_ALARM.format(spc_object.id))
         elif isinstance(spc_object, Zone):
             async_dispatcher_send(hass, SIGNAL_UPDATE_SENSOR.format(spc_object.id))
 
+    hass.data.setdefault(DOMAIN, {})
+
     session = aiohttp_client.async_get_clientsession(hass)
 
-    spc = SpcWebGateway(
+    client = SpcWebGateway(
         loop=hass.loop,
```

Comment:
Just a note that since (python 3.6 I think) you can get the loop without passing it into the class